tests/epyccel/test_arrays.py

```diff
@@ -3975,6 +3975,808 @@ def test_array_ndmin_2_order(language):
     check_array_equal(f1(d), f2(d))
     check_array_equal(f1(e), f2(e))
 
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_bool_to_other_types_1(language):
+    size = (2, 2)
+    bl = randint(0, 2, size = size, dtype= bool)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int8, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_int8(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_bool, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_bool(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int16, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_int16(bl)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_bool_to_other_types_2(language):
+    size = (2, 2)
+    bl = randint(0, 2, size = size, dtype= bool)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int32, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_int32(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int64, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_int64(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_float32, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_float32(bl)
```

Comment:
Please can you explain the logic for how these functions are split up? Did you just want to avoid having a large test?

Reply:
Putting everything under one function was making that function to execute for a long time. It was clearly visible when I ran the run_tests_py3.sh script. So i have split them them into 4 functions for each datatype.

Reply:
This is probably linked to my comment below. Translating/compiling is slow so if you group by argument type instead of result type then yes the tests will be significantly slower